src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -81,7 +83,7 @@
  * 
  * @author Kohsuke Kawaguchi
  */
-@Mojo(name="run")
+@Mojo(name="run", requiresDependencyResolution=ResolutionScope.TEST)
```

Comment:
I think *something* is attaching artifacts to the reactor and then they get wiped by clean and the re-attach is appending to the list, not replacing... does it happen with Maven 3.2.x?